gen/common/util.ml

```diff
@@ -61,14 +61,78 @@ let parse_cmdline options get_cmd_arg =
     get_cmd_arg
     (sprintf "Usage %s [options] [arg]*\noptions are:" Sys.argv.(0))
 
+module type Monad = sig
+  type 'a t
+  val pure : 'a -> 'a t
+  val bind : 'a t -> ('a -> 'b t) -> 'b t
+
+  module Infix : sig
+    val (>>=) : 'a t -> ('a -> 'b t) -> 'b t
+    val (let*) : 'a t -> ('a -> 'b t) -> 'b t
```

Comment:
These two functions `(>>=)` and `(let*)` are always the same?

Reply:
They are expected to be implemented in the same exact way, yes. With the difference that `let*` is intended to be used to implement monadic binding:
```
let* x = foo in
bar x
```
vs
```
foo >>= fun x -> bar x
```
Does that answer your question?